pkg/controllers/clusterresourceplacement/controller.go

```diff
@@ -206,15 +206,44 @@ func (r *Reconciler) handleUpdate(ctx context.Context, crp *fleetv1beta1.Cluster
 		klog.V(2).InfoS("Placement has been scheduled", "clusterResourcePlacement", crpKObj, "generation", crp.Generation)
 		r.Recorder.Event(crp, corev1.EventTypeNormal, "PlacementScheduleSuccess", "Successfully scheduled the placement")
 	}
+
+	if r.UseNewConditions {
+		// We skip checking the last resource condition (available) because it will be covered by checking isRolloutCompleted func.
+		for i := RolloutStartedCondition; i < TotalCondition-1; i++ {
+			oldCond := oldCRP.GetCondition(string(i.ClusterResourcePlacementConditionType()))
+			newCond := crp.GetCondition(string(i.ClusterResourcePlacementConditionType()))
+			if !condition.IsConditionStatusTrue(oldCond, oldCRP.Generation) &&
+				condition.IsConditionStatusTrue(newCond, crp.Generation) {
+				klog.V(2).InfoS("Placement resource condition status has been changed to true", "clusterResourcePlacement", crpKObj, "generation", crp.Generation, "condition", i.ClusterResourcePlacementConditionType())
+				r.Recorder.Event(crp, corev1.EventTypeNormal, i.EventReasonForTrue(), i.EventMessageForTrue())
+			}
+		}
+
+		// There is no need to check if the CRP is available or not.
+		// If the available condition is true, it means the rollout is completed.
+		if isRolloutCompleted(r.UseNewConditions, crp) {
+			if !isRolloutCompleted(r.UseNewConditions, oldCRP) {
+				klog.V(2).InfoS("Placement rollout has finished and resources are available", "clusterResourcePlacement", crpKObj, "generation", crp.Generation)
+				r.Recorder.Event(crp, corev1.EventTypeNormal, "PlacementRolloutCompleted", "Resources are available in the selected clusters")
+			}
+			// We don't need to requeue any request now by watching the binding changes
+			return ctrl.Result{}, nil
```

Comment:
> where is the change to watch binding?

It will be a separate PR, we need to add a binding watcher for the crp.

> I assume that needs to be tested in at least integration test?
Will do